Failing doctest CI Job: `The current Dask DataFrame implementation is deprecated.`

### What happened?

The [doctest CI job for my Pull Request](https://github.com/pydata/xarray/actions/runs/7854959732/job/21436224544?pr=8698) failed. The failure seems at first glance to be unrelated to my code changes. It seems related to a Dask warning.

Note: I create this issue for logging purposes ; it might become relevant only once another unrelated PR is subject to the same bug.

### What did you expect to happen?

I expected the `doctest` CI Job to pass. This error happens both on the online CI and locally when running 

```
python -m pytest --doctest-modules xarray --ignore xarray/tests --ignore xarray/datatree_ -Werror
```

(the command is taken from the CI definition file: https://github.com/pydata/xarray/actions/runs/7854959732/workflow?pr=8698#L83)

### Minimal Complete Verifiable Example

```Python
N/A
```


### MVCE confirmation

- [ ] Minimal example — the example is as focused as reasonably possible to demonstrate the underlying issue in xarray.
- [ ] Complete example — the example is self-contained, including all data and the text of any traceback.
- [ ] Verifiable example — the example copy & pastes into an IPython prompt or [Binder notebook](https://mybinder.org/v2/gh/pydata/xarray/main?urlpath=lab/tree/doc/examples/blank_template.ipynb), returning the result.
- [ ] New issue — a search of GitHub Issues suggests this is not a duplicate.
- [ ] Recent environment — the issue occurs with the latest version of xarray and its dependencies.

### Relevant log output

```Python
=================================== FAILURES ===================================
_________ [doctest] xarray.core.dataarray.DataArray.to_dask_dataframe __________
7373         ...     dims=("time", "lat", "lon"),
7374         ...     coords={
7375         ...         "time": np.arange(4),
7376         ...         "lat": [-30, -20],
7377         ...         "lon": [120, 130],
7378         ...     },
7379         ...     name="eg_dataarray",
7380         ...     attrs={"units": "Celsius", "description": "Random temperature data"},
7381         ... )
7382         >>> da.to_dask_dataframe(["lat", "lon", "time"]).compute()
UNEXPECTED EXCEPTION: DeprecationWarning("The current Dask DataFrame implementation is deprecated. \nIn a future release, Dask DataFrame will use new implementation that\ncontains several improvements including a logical query planning.\nThe user-facing DataFrame API will remain unchanged.\n\nThe new implementation is already available and can be enabled by\ninstalling the dask-expr library:\n\n    $ pip install dask-expr\n\nand turning the query planning option on:\n\n    >>> import dask\n    >>> dask.config.set({'dataframe.query-planning': True})\n    >>> import dask.dataframe as dd\n\nAPI documentation for the new implementation is available at\n[https://docs.dask.org/en/stable/dask-expr-api.html\n\nAny](https://docs.dask.org/en/stable/dask-expr-api.html/n/nAny) feedback can be reported on the Dask issue tracker\nhttps://github.com/dask/dask/issues \n")
Traceback (most recent call last):
  File "/home/runner/micromamba/envs/xarray-tests/lib/python3.11/doctest.py", line 1353, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest xarray.core.dataarray.DataArray.to_dask_dataframe[1]>", line 1, in <module>
  File "/home/runner/work/xarray/xarray/xarray/core/dataarray.py", line 7408, in to_dask_dataframe
    return ds.to_dask_dataframe(dim_order, set_index)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/xarray/xarray/xarray/core/dataset.py", line 7369, in to_dask_dataframe
    import dask.dataframe as dd
  File "/home/runner/micromamba/envs/xarray-tests/lib/python3.11/site-packages/dask/dataframe/__init__.py", line 162, in <module>
    warnings.warn(
DeprecationWarning: The current Dask DataFrame implementation is deprecated. 
In a future release, Dask DataFrame will use new implementation that
contains several improvements including a logical query planning.
The user-facing DataFrame API will remain unchanged.

The new implementation is already available and can be enabled by
installing the dask-expr library:

    $ pip install dask-expr

and turning the query planning option on:

    >>> import dask
    >>> dask.config.set({'dataframe.query-planning': True})
    >>> import dask.dataframe as dd

API documentation for the new implementation is available at
https://docs.dask.org/en/stable/dask-expr-api.html

Any feedback can be reported on the Dask issue tracker
https://github.com/dask/dask/issues 
/home/runner/work/xarray/xarray/xarray/core/dataarray.py:7382: UnexpectedException
=========================== short test summary info ============================
FAILED xarray/core/dataarray.py::xarray.core.dataarray.DataArray.to_dask_dataframe
============= 1 failed, 301 passed, 2 skipped in 78.04s (0:01:18) ==============
Error: Process completed with exit code 1.
```


### Anything else we need to know?

_No response_

### Environment

N/A

--- a/ci/requirements/environment.yml
+++ b/ci/requirements/environment.yml
@@ -9,6 +9,7 @@ dependencies:
   - cartopy
   - cftime
   - dask-core
+  - dask-expr # dask raises a deprecation warning without this, breaking doctests
   - distributed
   - flox
   - fsspec!=2021.7.0
@@ -32,7 +33,7 @@ dependencies:
   - pip
   - pooch
   - pre-commit
-  - pyarrow # pandas makes a deprecation warning without this, breaking doctests
+  - pyarrow # pandas raises a deprecation warning without this, breaking doctests
   - pydap
   - pytest
   - pytest-cov

--- a/conftest.py
+++ b/conftest.py
@@ -39,3 +39,11 @@ def add_standard_imports(doctest_namespace, tmpdir):
 
     # always switch to the temporary directory, so files get written there
     tmpdir.chdir()
+
+    # Avoid the dask deprecation warning, can remove if CI passes without this.
+    try:
+        import dask
+    except ImportError:
+        pass
+    else:
+        dask.config.set({"dataframe.query-planning": True})

--- a/conftest.py
+++ b/conftest.py
@@ -46,4 +46,4 @@ def add_standard_imports(doctest_namespace, tmpdir):
     except ImportError:
         pass
     else:
-        dask.config.set({"dataframe.query-planning": False})
+        dask.config.set({"dataframe.query-planning": True})

--- a/conftest.py
+++ b/conftest.py
@@ -46,4 +46,4 @@ def add_standard_imports(doctest_namespace, tmpdir):
     except ImportError:
         pass
     else:
-        dask.config.set({"dataframe.query-planning": True})
+        dask.config.set({"dataframe.query-planning": False})

